src/Orleans.Runtime/Facet/Persistent/PersistentStateStorageFactory.cs

```diff
@@ -25,22 +25,21 @@ public IPersistentState<TState> Create<TState>(IGrainActivationContext context,
             return bridge;
         }
 
-        protected virtual string GetFullStateName(IGrainActivationContext context, IPersistentStateConfiguration cfg)
+        protected virtual string GetFullStateName(IGrainContext context, IPersistentStateConfiguration cfg)
         {
-            return $"{RuntimeTypeNameFormatter.Format(context.GrainType)}.{cfg.StateName}";
+            return $"{context.GrainId}.{cfg.StateName}";
```

Comment:
@ReubenBond are you sure this is correct? I'm working on a custom storage provider and was trying to figure out how the 'grain type' value was being constructed and noticed this changed since the latest release. Currently, grain types are fully formatted types that can be parsed by `Type.GetType()` + the state name, but here it seems you're changing the type to a grain ID, which would at the least be backwards incompatible. Maybe I'm missing something.

Reply:
You're right. I don't have a final solution for that yet, but I was considering we would want to either fork providers or provide an adapter in order to support migration from grains using legacy ids to the simplified id system.

I would also be ok with a PR to revert this particular change, if that helps your case.

We will need to audit the changes before final release to make sure that a migration path is viable.

Note that this code has not been included in any releases - all releases are made from 3.x.x tags and do not include these commits.